src/tui/state.rs

```diff
@@ -91,6 +91,10 @@ impl<'a> State<'a> {
         state.handle_tab()?;
         Ok(state)
     }
+    
+    pub fn change_analyzer(&mut self, analyzer: Analyzer<'a>) {
```

Comment:
`analyzer` is a public field, do we need this function? (if so, let's rename it to `update_analyzer`)

Reply:
Agreed, i think more consistently will be remove this